ws/src/server.rs

```diff
@@ -74,7 +74,7 @@ impl Server {
 			config.out_buffer_capacity_hard_limit = max_out_buffer_capacity;
 			// don't grow non-final fragments (to prevent DOS)
 			config.fragments_grow = false;
-			config.fragments_capacity = cmp::max(1, max_payload_bytes / config.fragment_size);
+			config.fragments_capacity = cmp::min(cmp::max(1, max_payload_bytes / config.fragment_size), 4096);
```

Comment:
Isn't 4096 too little? Since fragments cannot grow, with 4096 fragments and `fragment_size` set to default we will be able to handle fragmented payload size up to `4096 * 65535 ~= 255MB`. 
Maybe better change `max_payload_bytes` to be an optional parameter (`None` meaning no limit and growing fragments) and have it fail early with a helpful error message in case `max_payload_bytes` is unreasonably big (bigger than the amount of RAM or sth)?

Reply:
I'm not arguing about sensibility of the limit. I agree that for our use cases 256MB of request size is probably more than enough.
I am against making Polkadot/Substrate specific assumptions in the general `jsonrpc` library we are building here. Silently capping the max amount of bytes in a request, despite having a separate configuration option (`max_payload_bytes`) is IMHO incorrect.
I'd suggest to simply error out if `max_payload_bytes` is configured for something above `255MB` and keeping the behavior introduced in this PR or (IMHO more in the vein of "general library") making it more flexbile by enabling growing fragments.

Reply:
You do make a good point.

Okay, so I've made it so that if the fragments' capacity is over the limit it just lets them grow.